src/pages/index.js

```diff
@@ -47,13 +88,73 @@ function fillProfileForm() {
 /*---------------------------------------------------------------------------------------------------------*/
 /*                                                Event Handlers                                           */
 /*---------------------------------------------------------------------------------------------------------*/
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
+);
+
+function handleProfileEditSubmit(userData) {
+  formProfileEditModal.setLoading(true);
+  api
+    .updateUserProfile(userData)
+    .then(() => {
+      userInfo.setUserInfo(userData.name, userData.about);
+      formProfileEditModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+
+    .finally(() => {
+      formProfileEditModal.setLoading(false);
+    });
+}
 
-function handleProfileEditSubmit(formData) {
-  userInfo.setUserInfo(formData.title, formData.description);
-  formProfileEditModal.close();
+function handleLikeClick(card) {
+  if (card.isLiked) {
+    api
+      .dislikeCard(card._id)
+      .then(() => {
+        card.handleLikeButton();
+      })
+      .catch((err) => console.error(err));
+  } else {
+    api
+      .likeCard(card._id)
+      .then(() => {
+        card.handleLikeButton();
+      })
+      .catch((err) => console.error(err));
+  }
 }
 
+function handleImageClick(imageData) {
+  cardPreviewImageModal.open(imageData);
+}
+
+const deleteCardPopup = new PopupWithConfirmation(selectors.deleteCardPopup);
+
+function handleDeleteButton(cardID, card) {
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitAction(() => {
+    api
```

Comment:
Needs correction: you should also notify the user about loading by changing the submit button text

Reply:
I updated the code so that users are notified about loading.